app/scripts/migrations/063.js

```diff
@@ -0,0 +1,55 @@
+import { cloneDeep } from 'lodash';
+
+const version = 63;
+
+/**
+ * Moves token state from preferences controller to TokensController
+ */
+export default {
+  version,
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData);
+    versionedData.meta.version = version;
+    const state = versionedData.data;
+    const newState = transformState(state);
+    versionedData.data = newState;
+    return versionedData;
+  },
+};
+
+function transformState(state) {
+  const accountTokens = state?.PreferencesController?.accountTokens;
+  const accountHiddenTokens = state?.PreferencesController?.accountHiddenTokens;
+
+  if (accountTokens || accountHiddenTokens) {
+    const flattenedIgnoredTokens = [];
```

Comment:
So the ignoredTokens data in the TokensController is currently flat. This makes ignored tokens ignored across all chains and accounts which is probably not ideal... I had made it this way in the TokensController because this was the shape in the AssetsController. Perhaps we should make the TokensController version conform to the current shape of accountHiddenTokens and change mobile to use that pattern instead of what I'm doing here? @danjm

Reply:
Yes, I think you are right.

Reply:
Might be worth asking mobile if there is any reason we don't do that already. But the new token detection will detect tokens by network, so it makes sense to ignore by network.

Reply:
@danjm Here is a PR to update the shape in the TokensController/AssetsDetectionController so this can move forward: https://github.com/MetaMask/controllers/pull/570